huma.go

```diff
@@ -1293,12 +1293,22 @@ func Register[I, O any](api API, op Operation, handler func(context.Context, *I)
 								})
 							}
 						} else {
-							// Set defaults for any fields that were not in the input.
+							// Set defaults for any fields that were not in the input:
+							atLeastOneFieldIsOverridden := false
 							defaults.Every(v, func(item reflect.Value, def any) {
-								if item.IsZero() {
+								if item.IsZero() && !reflect.ValueOf(def).IsZero() {
+									// There is one caveat here:
+									//     - IF the default value is not the 'zero' of the field type
+									//     - AND the 'zero' of the field type is a valid value specified in the input
+									//     - THEN the input required field value is overridden with a different value
 									item.Set(reflect.Indirect(reflect.ValueOf(def)))
+									atLeastOneFieldIsOverridden = true
 								}
 							})
+							// Restore wrongly overridden input values if any
+							if atLeastOneFieldIsOverridden {
+								_ = api.Unmarshal(ctx.Header("Content-Type"), body, f.Addr().Interface())
```

Comment:
This seems kind of inefficient and I'm wondering if there are any edge cases where it would inadvertently overwrite something... 🤔

Reply:
[Here](https://github.com/danielgtaylor/huma/pull/628) I propose a different way to fix this.